robolectric/src/test/java/org/robolectric/shadows/ShadowNotificationManagerTest.java

```diff
@@ -93,4 +97,32 @@ public void testCancelAll() throws Exception {
     assertNull(shadowOf(notificationManager).getNotification(null, 1));
     assertNull(shadowOf(notificationManager).getNotification(null, 31));
   }
+
+  @Test
+  public void testGetActiveNotifications() throws Exception {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+      // This API was added in M.
+      return;
+    }
+    notificationManager.notify(1, notification1);
+    notificationManager.notify(31, notification2);
+
+    StatusBarNotification[] statusBarNotifications =
+        shadowOf(notificationManager).getActiveNotifications();
+    assertEquals(2, statusBarNotifications.length);
+    boolean hasNotification1 = false;
+    boolean hasNotification2 = false;
+    for (StatusBarNotification notification : statusBarNotifications) {
```

Comment:
Actually, assertj offers an arguably cleaner way to do this. Take a look and let me know if you agree. I can also make an explicit Extractor to grab the notification instead of using extractingResultOf if you'd prefer this fail at compile time instead of runtime should the getNotification method be removed for some reason (though this would mean removing public API, which happens extremely rarely).